### DISCUSSION TITLE
[Logs Explore] Allow Displaying Timestamps with Millisecond Level

### DISCUSSION THREAD

**Discussion by sherifkayad** [grafana/grafana#45120] [Dashboards, Panels & Visualizations]
Currently the log explorer (with Loki e.g.) is only displaying times up to the second .. As an illustration:
![image](https://user-images.githubusercontent.com/8362784/153155695-5ee06d51-dbda-40f9-975f-101d78947330.png)

Would be great to show the timestamps in milliseconds (e.g. `2022-02-09T08:21:59.098Z`) .. Note that the timestamp is already available as part of the `ts` field on the log.
![image](https://user-images.githubusercontent.com/8362784/153156038-82b3fd6f-da73-4ec8-b40e-5c876d57bb26.png)

Here we could think of enabling this using a toggle on the explore level or so (similar to the toggles `Time`, `Prettify JSON`, .. etc.)

**Comment by jtackaberry** [comment]
Even better would be a configurable timestamp precision.

But I'd happily live with milliseconds -- second granularity gives Loki a toy-like impression (actual feedback from a developer I demoed Loki to), and it renders logs downloaded via Grafana much less useful for analysis than they could be.

**Comment by svennergr** [comment]
Hey folks,
we implemented milliseconds in https://github.com/grafana/grafana/pull/64372.

**Reply by svennergr** [reply]
Short addition, I added a FU discussion about making the time format configurable: https://github.com/grafana/grafana/discussions/64389
